x/finality/keeper/power_dist_change.go

```diff
@@ -332,6 +340,14 @@ func (k Keeper) ProcessAllPowerDistUpdateEvents(
 		newFP := k.loadFP(ctx, fpByBtcPkHex, fpBTCPKHex)
```

Comment:
I think you are righ about about in above comments (https://github.com/babylonlabs-io/babylon/pull/371/files#r1896052253)

Though one thing that worries me is that we should probably ignore fp here if he is already slashed as even though we do not accept delegations for slashed finality providers, it is still possible to generate voting power events for them:
- it is still possible to activate delegation which was created without inclusion proof
- it is sittl possible to unbond delegation which would create unbonding event

Could you try to add some tests for this cases to verify that it is handled properly currently and add fixe if they dont ? (basically we do not want for slashed fp to ever end up in voting power set ever again)

Reply:
I am verifying if they are slashed, in the creation of `activedSatsByFpBtcPk` list, but yeah adding the check as Gai [suggested](https://github.com/babylonlabs-io/babylon/pull/371#pullrequestreview-2521909396) seems to make a lot of sense to me